plugins/module_utils/network/routeros/routeros.py

```diff
@@ -66,9 +66,12 @@ def get_capabilities(module):
     if hasattr(module, '_routeros_capabilities'):
         return module._routeros_capabilities
 
-    capabilities = Connection(module._socket_path).get_capabilities()
-    module._routeros_capabilities = json.loads(capabilities)
-    return module._routeros_capabilities
+    try:
+        capabilities = Connection(module._socket_path).get_capabilities()
+        module._routeros_capabilities = json.loads(capabilities)
+        return module._routeros_capabilities
+    except ConnectionError as exc:
+        module.fail_json(msg=to_text(exc))
```

Comment:
```suggestion
        module.fail_json(msg=to_native(exc))
```
https://docs.ansible.com/ansible/latest/dev_guide/developing_plugins.html#raising-errors

Reply:
to_text is used in different places on exceptions handling over the code, that could be an opportunity to refactor all the occurences, what do you think?

Reply:
I think this would be better suited for a follow-up PR. Makes it easier to backport this one (since that currently has to be done pretty much manually).

Reply:
Added the to_native module, and changed the msg to: 
`` to_native(exc, errors='surrogate_then_replace'))`` 
for consistence.